etc/pachd/Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM debian:buster
```

Comment:
Drive-by comment: is it deliberate that we're basing the `pachd` image on `debian` from now on or is this possibly a debugging artifact? (I think it was `scratch` before: https://github.com/pachyderm/pachyderm/blob/master/Dockerfile.pachd#L1)

Reply:
Scratch should work fine, I can switch to that. I was using `debian:buster` because to `go` image used for building is built on top of debian buster.

Reply:
Also, drive-by comments/reviews are always appreciated! Especially for something like this since it's so cross-cutting.

Reply:
Glad it was helpful :-). I was just curious how this change worked, because I remembered we'd talked about it at the onsite and I was impressed you managed to refactor it this fast.

Reply:
I'm not sure why, but switching to scratch causes nearly all of the tests to fail

Reply:
With this setup, it looks like worker containers can't start up when they're based off of scratch. It can't start the worker script (`/pach/worker.sh`) because it can't find the script:

```
k logs pod/pipeline-testsimplepipelinec93e65aea6d5-v1-pzc7t init
standard_init_linux.go:211: exec user process caused "no such file or directory"
```

I'm not exactly sure how to debug this further, because scratch doesn't even have a shell I can pop into to debug.

Reply:
Ah, `worker.sh` is a shell script, but scratch has no shell to run it with. It turns out that the [master branch's worker is built off of ubuntu 16](https://github.com/pachyderm/pachyderm/blob/1e371b7ab20cd191bb8b1c16fecb63464bca51ee/Dockerfile.worker#L1). Given that this is quite a bit more heavy-weight than `debian:buster`, I'm inclined to use `debian:buster` for both pachd and worker:

* The worker will be lighter-weight than what we have on master
* It'll ensure pachd is easily debuggable in case things go wrong

What do you think?

Reply:
Ah, I'm wrong about debian images being smaller - they are, but it's pretty marginal.

Reply:
Have you considered alpine or debian:buster-slim?  Not sure how alpine's use of musl instead of glibc would affect us.

Reply:
Not yet, but that's on the to-try list. It's either that or just ripping out our very minimal shell usage and sticking to scratch.

Alpine's use of musl shouldn't affect us, because these containers are just running go code with cgo disabled (well, minus the one shell script that worker calls.) Go binaries are statically linked, and implement their own syscall interface rather than using libc.